src/cpp/rediscluster.cpp

```diff
@@ -322,39 +359,34 @@ CommandReply RedisCluster::set_model(const std::string& model_name,
                                      const std::vector<std::string>& inputs,
                                      const std::vector<std::string>& outputs)
 {
-    std::vector<DBNode>::const_iterator node = _db_nodes.cbegin();
+    // Build the basic command
     CommandReply reply;
-    for ( ; node != _db_nodes.cend(); node++) {
-        // Build the node prefix
-        std::string prefixed_key = "{" + node->prefix + "}." + model_name;
-
-        // Build the MODELSET commnd
-        CompoundCommand cmd;
-        cmd << "AI.MODELSTORE" << Keyfield(prefixed_key) << backend << device;
+    AddressAllCommand cmd;
+    cmd.key_index = 1;
+    cmd << "AI.MODELSTORE" << Keyfield(model_name) << backend << device;
 
-        // Add optional fields as requested
-        if (tag.size() > 0) {
-            cmd << "TAG" << tag;
-        }
-        if (batch_size > 0) {
-            cmd << "BATCHSIZE" << std::to_string(batch_size);
-        }
-        if (min_batch_size > 0) {
-            cmd << "MINBATCHSIZE" << std::to_string(min_batch_size);
-        }
-        if ( inputs.size() > 0) {
-            cmd << "INPUTS" << std::to_string(inputs.size()) << inputs;
-        }
-        if (outputs.size() > 0) {
-            cmd << "OUTPUTS" << std::to_string(outputs.size()) << outputs;
-        }
-        cmd << "BLOB" << model;
+    // Add optional fields as requested
```

Comment:
Are the optional arguments order dependent at all? Also, gotta say it makes it so nice to build the command via `<<`

Reply:
Yes, the order is prescribed